### PR TITLE
Add the rack-mini-profiler

### PR DESCRIPTION
**Pull request by philbogle**
The rack-mini-profiler optionally displays a "speed badge" for every page (in development only) showing how long each part of the page took to render and where the time went.  You can see the number of SQL queries run and drill in to see exactly which queries were run and the corresponding stack traces.

To ensure there is no impact on production, the profiler gem is included only in development mode and profiling is off by default.  To enable the profiler,  set the environment variable RACK_MINI_PROFILER=on before starting your server.  Alternatively, you can enable the profiler dynamically by adding to "pp=enabled" to the query string and then refreshing the page one more time - don't forget the extra refresh.  "pp=help" will display additional options.

Below is an example of the speed badge for a HOC page:
image

Clicking into a particular resource on the badge shows a breakdown of where the time went, including 2 sql queries: 
image

Clicking into the SQL queries shows details on each query:
image

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -140,7 +141,9 @@ gem 'twilio-ruby' # SMS API for send-to-phone feature
 gem 'font-awesome-rails'
 gem 'sequel', '~> 4.10.0'
 gem 'user_agent_parser'
-gem 'heroku_rails_deflate' # gzip rails content and static assets
+
+gem 'heroku_rails_deflate', group: [:staging, :production] # gzip rails content and static assets
```
Comment (joshlory) [inline Gemfile]: Do we want this on `:test` and `:levelbuilder` as well?

File: dashboard/config/initializers/mini_profiler.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+if Rails.env.development?
+  require 'rack-mini-profiler'
+
+  # Configure the rack mini-profiler, which displays a on-page breakdown of the
+  # time spent rendering the page including details on the SQL queries run.
+
+  # Display the mini-profiler on the right side of the page so that it is likely
```
Comment (joshlory) [inline dashboard/config/initializers/mini_profiler.rb]: "less likely"? :smile:

File: dashboard/app/controllers/application_controller.rb [hunk 3]
```diff
@@ -26,6 +26,16 @@ def fix_crawlers_with_bad_accept_headers
     end
   end
 
+  if Rails.env.development?
+    # Enable or disable the rack mini-profiler if the 'pp' query string parameter is set.
+    # pp='disabled' will disable it; any other value will enable it.
+    before_filter :maybe_enable_profiler
+    def maybe_enable_profiler
+      pp = params['pp']
+      ENV['RACK_MINI_PROFILER'] = (pp == 'disabled') ? 'off' : 'on' if pp
```
Comment (joshlory) [inline dashboard/app/controllers/application_controller.rb]: Nit: an `if` and ternary on the same line gets a little convoluted.  Cleaner to write as this?
```ruby
if pp
  ENV['RACK_MINI_PROFILER'] = (pp == 'disabled') ? 'off' : 'on'
end
```

**Comment by joshlory** [comment]
Very cool gem!  LGTM

File: dashboard/config/initializers/mini_profiler.rb (hunk 2 above)
Comment (philbogle) [inline dashboard/config/initializers/mini_profiler.rb]: Fixed.

File: Gemfile (hunk 1 above)
Comment (philbogle) [inline Gemfile]: Yes, done.

**Comment by philbogle** [comment]
Thanks for the review Josh.

**Pull request merged**